pkg/services/ngalert/alert_definition.go

```diff
@@ -0,0 +1,13 @@
+package ngalert
+
+// preSave sets datasource and loads the updated model for each alert query.
+func (alertDefinition *AlertDefinition) preSave() error {
+	for i, q := range alertDefinition.Data {
+		err := q.PreSave(alertDefinition.OrgId)
+		if err != nil {
+			return err
```

Comment:
I think we should wrap the error with the RefID of the alert query in this loop so when there are problems you can see which query has the issue. (or perhaps i and queryRefId)